modules/programs/gitui/gitui.nix

```diff
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.gitui;
+
+in {
+  meta.maintainers = [ hm.maintainers.mifom ];
+
+  options.programs.gitui = {
+    enable =
+      mkEnableOption "gitui, blazing fast terminal-ui for git written in rust";
+
+    keyConfig = mkOption {
+      type = types.either types.path types.lines;
+      default = ./default_key_config.ron;
+      example = literalExpression '''''
+        exit: ( code: Char('c'), modifiers: ( bits: 2,),),
+        quit: ( code: Char('q'), modifiers: ( bits: 0,),),
+        exit_popup: ( code: Esc, modifiers: ( bits: 0,),),
+      ''''';
+      description = ''
+        Key config in Ron file format. This is written to
+        <filename>$XDG_CONFIG_HOME/gitui/key_config.ron</filename>.
+      '';
+    };
+    theme = mkOption {
+      type = types.either types.path types.lines;
+      default = ./default_theme.ron;
+      description = ''
+        Theme in Ron file format. This is written to
+        <filename>$XDG_CONFIG_HOME/gitui/theme.ron</filename>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.gitui ];
```

Comment:
Can you make the package configurable?

```
options.programs.gitui.package = mkOption {
  type = types.package;
  default = pkgs.gitui;
  defaultText = "pkgs.gitui";
  description = "The package to use.";
}
```

Reply:
https://github.com/nix-community/home-manager/pull/2446/commits/5ad2a4ad96dfddad12ced599464bd2920b5fd491